kythe/go/serving/explore/explore_test.go

```diff
@@ -0,0 +1,478 @@
+	"kythe.io/kythe/go/test/testutil"
+
+	"github.com/golang/protobuf/proto"
+
+	epb "kythe.io/kythe/proto/explore_go_proto"
+	srvpb "kythe.io/kythe/proto/serving_go_proto"
+)
+
+var (
+	ctx = context.Background()
+
+	p1       = "kythe:#parent1"
+	p2       = "kythe:#parent2"
+	p1c1     = "kythe:#p1child1"
+	p1c2     = "kythe:#p1child2"
+	badType  = "kythe:#baddatatype"
+	dontcare = "kythe:#dontcare"
+	f1       = "kythe:#function1"
+	f2       = "kythe:#function2"
+	f1r1     = "kythe:#f1caller1"
+	fr       = "kythe:#fcaller"
+	f2r1     = "kythe:#f2caller1"
+	f3       = "kythe:#function3_recursive"
+
+	parentToChildren = &protoTable{
+		p1: &srvpb.Relatives{
+			Tickets: []string{p1c1, p1c2},
+			Type:    srvpb.Relatives_CHILDREN,
+		},
+		badType: &srvpb.Relatives{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Relatives_PARENTS,
+		},
+	}
+
+	childToParents = &protoTable{
+		p1c1: &srvpb.Relatives{
+			Tickets: []string{p1},
+			Type:    srvpb.Relatives_PARENTS,
+		},
+		p1c2: &srvpb.Relatives{
+			Tickets: []string{p2},
+			Type:    srvpb.Relatives_PARENTS,
+		},
+		badType: &srvpb.Relatives{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Relatives_CHILDREN,
+		},
+	}
+
+	functionToCallers = &protoTable{
+		f1: &srvpb.Callgraph{
+			Tickets: []string{f1r1, fr},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		f2: &srvpb.Callgraph{
+			Tickets: []string{f2r1, fr},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		f3: &srvpb.Callgraph{
+			Tickets: []string{f1, fr, f3},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		fr: &srvpb.Callgraph{
+			Tickets: []string{f2},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		badType: &srvpb.Callgraph{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+	}
+
+	functionToCallees = &protoTable{
+		f1: &srvpb.Callgraph{
+			Tickets: []string{f3},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f2: &srvpb.Callgraph{
+			Tickets: []string{fr},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f3: &srvpb.Callgraph{
+			Tickets: []string{f3},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		fr: &srvpb.Callgraph{
+			Tickets: []string{f1, f2, f3},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f1r1: &srvpb.Callgraph{
+			Tickets: []string{f1},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f2r1: &srvpb.Callgraph{
+			Tickets: []string{f2},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		badType: &srvpb.Callgraph{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+	}
+)
+
+func TestChildren_badData(t *testing.T) {
+	svc := Construct(t)
+
+	reply, err := svc.Children(ctx, &epb.ChildrenRequest{
+		Tickets: []string{badType},
+	})
+	if err == nil {
+		t.Errorf("Expected Children error for bad data, got: %v", reply)
+	}
+}
+
+func TestChildren_noData(t *testing.T) {
+	svc := Construct(t)
+
+	reply, err := svc.Children(ctx, &epb.ChildrenRequest{
+		Tickets: []string{"key with no data"},
+	})
+	if err != nil {
+		testutil.FatalOnErrT(t, "Children error: %v", err)
```

Comment:
`FatalOnErrT` does the `err != nil` check; remove the surrounding conditional.

Reply:
Done.